clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/data/task/jedis/JedisTaskRepository.groovy

```diff
@@ -48,10 +49,22 @@ class JedisTaskRepository implements TaskRepository {
 
   @Override
   Task create(String phase, String status) {
+    create(phase, status, UUID.randomUUID().toString())
+  }
+
+  @Override
+  Task create(String phase, String status, String key) {
+    String taskKey = TASK_KEY_MAP + ":" + key
+    def getExistingTask = jedis { it.get(taskKey) }
+    String existingTask = getExistingTask
+    if (existingTask) {
+      return get(existingTask)
+    }
     String taskId = jedis { it.incr('taskCounter') }
     def task = new JedisTask(taskId, System.currentTimeMillis(), this, false)
     addToHistory(new DefaultTaskStatus(phase, status, TaskState.STARTED), task)
     set(taskId, task)
+    jedis { it.append(taskKey, taskId) }
```

Comment:
The append at the end is not right (should be a `set`)

I think it would be better as more of a putIfAbsent equivalent to avoid the race condition of getExistingTask not existing at the start and two tasks kicking off:
````groovy
String taskKey = TASK_KEY_MAP + ":" + key
String taskId = jedis { it.incr('taskCounter') }
def task = new JedisTask(taskId, System.currentTimeMillis(), this, false)
addToHistory(new DefaultTaskStatus(phase, status, TaskState.STARTED), task)
set(taskId, task)
Long newTask = jedis { it.setnx(taskKey, taskId) }
if (newTask) {
 return task
}
//there's an existing taskId for this key, clean up what we just created and get the existing task:
addToHistory(new DefaultTaskStatus(phase, "Duplicate of $key", TaskState.FAILED), task)
String existingTask = jedis { it.get(taskKey) }
return get(existingTask)
````